src/codec2_math.h

```diff
@@ -0,0 +1,50 @@
+#ifndef CODEC2_MATH_H
+#define CODEC2_MATH_H
+
+//==========================================================================
+// Name:            codec2_math.h
+//
+// Purpose:         A wrapper around architecture specific math libraries 
+//                  used on embedded devices to improve Codec2 performance.
+// Created:         May 15, 2022
+// Authors:         Mooneer Salem
+//
+// License:
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License version 2.1,
+//  as published by the Free Software Foundation.  This program is
+//  distributed in the hope that it will be useful, but WITHOUT ANY
+//  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+//  License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program; if not, see <http://www.gnu.org/licenses/>.
+//
+//==========================================================================
+
+#if defined(__ARM_ARCH)
+#include "arm_math.h"
+
+#define codec2_dot_product_f32(left, right, len, result) (arm_dot_prod_f32(left, right, len, result))
+#define codec2_complex_dot_product_f32(left, right, len, resultReal, resultImag) (arm_cmplx_dot_prod_f32(left, right, len, resultReal, resultImag))
+
```

Comment:
Suggest the ARM case goes inside codec2_math.c, making the header file content just the function headers.

Reply:
Oh I see, there is no codec2_math.c by default, that's user provided :thinking:

Reply:
Hmm, so should we add platform specific implementation (like your ESP32 example above) in a codec_math.h?  That way other people can build codec2 on those platforms out of the box.

Reply:
> Hmm, so should we add platform specific implementation (like your ESP32 example above) in a codec_math.h? That way other people can build codec2 on those platforms out of the box.

At least for ESP32, some of the implementation of ESP-DSP (especially complex math as the dot product functions are real-only) doesn't translate over to a single function call so we probably can't unless it can be represented as an inline function. I'm not sure we necessarily want to support every single embedded architecture out there, either, though just simple wrappers to the platform's existing DSP library probably won't be a huge deal.